Refactor story repository hottest

nav-mike opened the pull request:
https://github.com/lobsters/lobsters/issues/1523

Move `hottest` method from the repository to Story model as a scope.

pushcx commented:
This is the big one for the home page, good to have some tests added for it. :)

I added a helper for testing while logged in and refactored the tests a bit. I didn't want to complain in your previous PRs, but I don't see enough value in the one-assertion-per-test style to justify the extra code and runtime. So in 1175f5bd I refactored them and I'd appreciate if you could follow this style in future StoryRepository refactorings. Thanks!

nav-mike commented:
> This is the big one for the home page, good to have some tests added for it. :)
> 
> I added a helper for testing while logged in and refactored the tests a bit. I didn't want to complain in your previous PRs, but I don't see enough value in the one-assertion-per-test style to justify the extra code and runtime. So in [1175f5b](https://github.com/lobsters/lobsters/commit/1175f5bdab9e05c90b8d3aef8e80dd826c91bd92) I refactored them and I'd appreciate if you could follow this style in future StoryRepository refactorings. Thanks!

Yeah, sure, no problem. Thx
